frontends/common/constantFolding.h

```diff
@@ -57,13 +58,13 @@ class ConstantFoldingPolicy {
  *      `IR::Declaration_Constant` nodes are initialized with
  *      compile-time known constants.
  */
-class DoConstantFolding : public Transform {
+class DoConstantFolding : public Transform, public ResolutionContext {
```

Comment:
Hmmm, what is the cost of invoking `ConstantFolding` now? Both ConstantFolding and StrengthReduction are useful passes for expression simplification without requiring a reference or type map.
https://github.com/p4lang/p4c/blob/main/frontends/p4/optimizeExpressions.h#L11

If this adds significant overhead this become a problem for interpreters which invoke these simplifiers a lot. Might have to duplicate the code here then.

Reply:
It can still be executed w/o refmap as before. No change here.